### PR TITLE
ci: add automation script

### PR DESCRIPTION
**Pull request by registerGen**
This automates the process described in `BUILD.md`.

Please take a look at <https://github.com/registerGen/lua-dev.nvim/actions/runs/2608229895>. Is there anything that doesn't work correctly?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: CI
+on:
+  push:
+    branches: [main]
```
Comment (folke) [inline .github/workflows/ci.yml:4]: This is awesome!

Just wondering if this won't trigger an indefinite loop?

Triggers on push to main, but the script itselfs pushes to main.

**Review by folke — CHANGES_REQUESTED** [review]
Looks good to me, except for the other comment

**Comment by folke** [comment]
I removed the `on` push to `main` and added a manual trigger. I've also kept the weekly.

Thanks again for making this!

**Comment by registerGen** [comment]
Thank you for reviewing this, too!

**Pull request merged**